honeycombio/terraform-honeycombio-buildevents-starter-pack
# Level up release workflow

## Which problem is this PR solving?

This PR is attempting to reduce the manual steps involved in releasing today.

This has a workflow that runs after every PR is merged that will keep a draft release up to date. Labels can be used to also manage versioning automatically. [Details](https://github.com/release-drafter/release-drafter)

**Ready to publish a new release?**

Workflow now also [documented here](https://github.com/honeycombio/terraform-honeycombio-buildevents-starter-pack/blob/e00efbef5c46972d1e0d186481fef9231025226b/RELEASING.md):

* Go over to the Releases and review the draft with notes already populated (make edits and save)

Example:
![image](https://user-images.githubusercontent.com/1179762/192903471-a6420b70-9b97-44a2-a3db-3b0fe87ffe9b.png)

* Locally run `make update-changelog`
* Open a PR and merge
* Go over to the Releases and Publish! (the new tag will be made - can be adjusted to the right commit at this point too)

## Short description of the changes

* Introducing release drafter to keep track of changes in a draft release; 
* adds Make task to allow for folks to update the version-controlled CHANGELOG.md file before release

## How to verify that this has the expected result

* Run `make update-changelog` locally to confirm the task can work
* Can validate the whole workflow with a release
